plugins/webp-uploads/settings.php

```diff
@@ -47,24 +73,56 @@ function webp_uploads_add_media_settings_field(): void {
 }
 add_action( 'admin_init', 'webp_uploads_add_media_settings_field' );
 
+/**
+ * Renders the settings field for the 'perflab_modern_image_format' setting.
+ *
+ * @since n.e.x.t
+ */
+function webp_uploads_generate_avif_webp_setting_callback(): void {
+
+	$selected       = webp_uploads_get_image_output_format();
+	$avif_supported = webp_uploads_mime_type_supported( 'image/avif' );
+	// Ensure WebP selected if AVIF is not supported.
+	if ( ! $avif_supported ) {
+		$selected = 'webp';
+	}
+	?>
+			<label for="perflab_modern_image_format">
+				<select name="perflab_modern_image_format" id="perflab_modern_image_format" aria-describedby="perflab_modern_image_format_description">
+					<option value="webp"<?php selected( 'webp', $selected ); ?>><?php esc_html_e( 'WebP', 'webp-uploads' ); ?></option>
+					<option value="avif"<?php selected( 'avif', $selected ); ?><?php disabled( ! $avif_supported ); ?>><?php esc_html_e( 'AVIF', 'webp-uploads' ); ?></option>
+				</select>
+				<?php esc_html_e( 'Generate images in this format', 'webp-uploads' ); ?>
+				</label>
+		<p class="description" id="perflab_modern_image_format_description"><?php esc_html_e( 'Select the format to use when generating new images from uploaded JPEGs.', 'webp-uploads' ); ?></p>
+			<?php
+			// Alert users if they lack AVIF support using a wp-admin notice.
+			if ( ! $avif_supported ) {
+				?>
+				<br />
+				<div class="notice notice-warning is-dismissible inline">
+					<p><b><?php esc_html_e( 'AVIF support is not available.', 'webp-uploads' ); ?></b></p>
+					<p><?php esc_html_e( 'AVIF support can only be enabled by your hosting provider, so contact them for more information.', 'webp-uploads' ); ?></p>
+				</div>
+				<?php
+			}
+			?>
```

Comment:
Looks like the formatting here got messed up a bit:

```suggestion
	<select name="perflab_modern_image_format" id="perflab_modern_image_format" aria-describedby="perflab_modern_image_format_description">
		<option value="webp"<?php selected( 'webp', $selected ); ?>><?php esc_html_e( 'WebP', 'webp-uploads' ); ?></option>
		<option value="avif"<?php selected( 'avif', $selected ); ?><?php disabled( ! $avif_supported ); ?>><?php esc_html_e( 'AVIF', 'webp-uploads' ); ?></option>
	</select>
	<label for="perflab_modern_image_format">
		<?php esc_html_e( 'Generate images in this format', 'webp-uploads' ); ?>
	</label>
	<p class="description" id="perflab_modern_image_format_description"><?php esc_html_e( 'Select the format to use when generating new images from uploaded JPEGs.', 'webp-uploads' ); ?></p>
	<?php if ( ! $avif_supported ) : ?>
		<br />
		<div class="notice notice-warning is-dismissible inline">
			<p><b><?php esc_html_e( 'AVIF support is not available.', 'webp-uploads' ); ?></b></p>
			<p><?php esc_html_e( 'AVIF support can only be enabled by your hosting provider, so contact them for more information.', 'webp-uploads' ); ?></p>
		</div>
	<?php endif;
```